Replace SyntaxHighlighter with Monaco Editor.

nojaf opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->

This replaces `react-syntax-highlighter` with `@monaco-editor/react`.


### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Use the magic "Fixes #1234" format, so the issues are -->
<!--- automatically closed when this PR is merged. -->

Fixes #3832.

The key motivation for us is to see large response bodies all the time.

### How Has This Been Tested?
<!--- Please describe in detail how you manually tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

I've been playing with this manually on my local machine.
The endpoint I'm testing is retrieving 2.8 MB of JSON and it gets instantly rendered in my browser.

### Screenshots (if appropriate):

![image](https://github.com/swagger-api/swagger-ui/assets/2621499/7d819db4-fb1d-4c46-a125-53ffc0679324)


## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

### My PR contains... 
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x] Dependency changes (any modification to dependencies in `package.json`)
- [?] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [x] are breaking changes to a public API (config options, System API, major UI change, etc).

I'm not quite sure yet what to do with the `syntaxHighlight.theme` setting. These themes were tied to `SyntaxHighlighter` and the Monaco editor has different themes out of the box.
Please let know how we can deal with this.

- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.

I'm not sure here: please tell me.

- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

As I'm new to this, I'm not quite sure what needs to be done here. Again, let me know and I'll look into it.

heldersepu commented:
https://raw.githack.com/nojaf/swagger-ui/large-bodies/dist/index.html#/pet/findPetsByStatus
I tested your version  and I'm not getting the same as your image...

heldersepu commented on test/unit/components/highlight-code.jsx:
```diff
@@ -18,13 +18,12 @@ describe("<HighlightCode />", () => {
     const props = {value: value, getConfigs: fakeGetConfigs}
     const wrapper = mount(<HighlightCode {...props} />)
-    const preTag = wrapper.find("pre")
+    const sectionTag = wrapper.find("section")
```
Nevermind ... I just realized this is just a test.

nojaf replied:
Yes, I believe `react-syntax-highlighter` created a `pre` tag and that was the assertion in the test.

heldersepu commented:
> https://raw.githack.com/nojaf/swagger-ui/large-bodies/dist/index.html#/pet/findPetsByStatus I tested your version and I'm not getting the same as your image...
> 
> the `<div class="highlight-code">` seem to be just a few pixels of height
> 
I think the issue here might be just old bundle, I added my changes to my branch and the height is fine:
https://raw.githack.com/heldersepu/swagger-ui/test/dist/index.html?docExpansion=None&defaultModelsExpandDepth=None#/pet/findPetsByStatus

heldersepu commented:
I change the theme to be dark and disabled the minimap to get as close as possible as it was:

![Screenshot from 2024-03-23 19-43-59](https://github.com/swagger-api/swagger-ui/assets/545474/0962f8d8-dfac-4dc0-9674-49772065c073)

simple changes:
https://github.com/swagger-api/swagger-ui/commit/45c763181bb968aa180f306cb1f68f7b5f8aebdf

nojaf commented:
Hi @heldersepu, thank you for the feedback!
I believe I addressed all your remarks.

Any thoughts on what to do with the `syntaxHighlight.theme` setting?

And could I get the workflow approval?
![image](https://github.com/swagger-api/swagger-ui/assets/2621499/d8c36672-983d-4f65-824c-05d164513bfa)

heldersepu commented:
@nojaf rebuild the bundles when you can I still can see the old version:
https://raw.githack.com/nojaf/swagger-ui/large-bodies/dist/index.html#/pet/findPetsByStatus 

would be nice to integrate the `syntaxHighlight.theme` into your changes

I'm not in control of the workflow some one from the team should do that

heldersepu commented:
@glowcloud & @char0n  👀

nojaf commented:
Hi @heldersepu, apologies for the delay. I took some time off.
I've rebuilt all bundles.

> would be nice to integrate the syntaxHighlight.theme into your changes

Happy to do this, but could use a pointer. Do we care about the existing values?

heldersepu commented:
I don't think the existing theme values are compatible with the Monaco editor 
test see what happens if an invalid theme is given, maybe there is a way we can provide a default or validate is one of the allowed themes

nojaf commented:
So, some e2e tests are failing like:

https://github.com/swagger-api/swagger-ui/blob/3e81a4f897e4be9530f3cc6b76e50570e3e61da4/test/e2e-cypress/e2e/bugs/4943.cy.js#L10-L19

![#4943 XML example not rendered correctly with oneOf -- should render oneOf property correctly (failed)](https://github.com/swagger-api/swagger-ui/assets/2621499/2c87728f-f8fa-433f-a527-46d12fd8f2b3)

The problem presumably is in the assertion and might not be an actual problem.

To run this: `npm run cy:start` and in a different terminal `npm run cy:run -- --spec "**/4943.cy.js"`.

Could I please get some pointers on how to address this?
I have never seen this Cypress thing up close and it is a tad overwhelming.

heldersepu commented:
@nojaf don't feel discouraged this PR is great, the syntax highlighter was in need of a major refactor...
The team must be busy with other priorities

heldersepu commented:
When you can address the:
This branch has conflicts that must be resolved

char0n commented:
Thanks for all the work on this, but we'll not be incorporating this major architectural change into core SwaggerUI. Due to https://github.com/swagger-api/swagger-ui/pull/9783, this can now be approached using the standard SwaggerUI plugin system. More information in https://github.com/swagger-api/swagger-ui/issues/3832#issuecomment-2047446217

I've also created a wiki page with your MonacoEditor plugin - https://github.com/swagger-api/swagger-ui/wiki/Plugins

char0n commented:
@nojaf it's experiments like these that push the evolution of SwaggerUI forward. Your PR did directly inspire https://github.com/swagger-api/swagger-ui/pull/9783
